src/Surface.cpp

```diff
@@ -166,20 +173,20 @@ void Surface::computeVerticesGrid(QSGGeometry::TexturedPoint2D* vertices, QSize
 
             if (dist > maxradius)
             {
-                x = center(0) + maxradius * cx / dist;
-                y = center(1) + maxradius * cy / dist;
+                x = static_cast<float>(center(0) + maxradius * cx / dist);
+                y = static_cast<float>(center(1) + maxradius * cy / dist);
             }
 
 
-            float u = i / (float)_subdivisions;
-            float v = j / (float)_subdivisions;
+            float u = fI / fSubdivisions;
+            float v = fJ / fSubdivisions;
```

Comment:
not sure but they could be const
```suggestion
            const float u = fI / fSubdivisions;
            const float v = fJ / fSubdivisions;
```

Reply:
Indeed they can!